### PR TITLE
added duplication check

### PR DESCRIPTION
**Pull request by david-deboer**
This addresses issue #130

### CODE REVIEW

File: hera_mc/cm_health.py [hunk 1]
```diff
@@ -14,11 +14,19 @@
 from . import cm_revisions
 
 
-def check_for_overlap(interval_i, interval_j):
-    if interval_j[0] <= interval_i[0]:
-        if interval_j[1] > interval_i[0]:
+def check_for_overlap(intervals):
```
Comment (bhazelton) [inline hera_mc/cm_health.py]: add a docstring -- what is the format of intervals?

File: hera_mc/cm_health.py [hunk 2]
```diff
@@ -42,7 +50,7 @@ def ensure_conndict(self):
             So all entries will have at least one pair.
         multiple - set - {key1, ...}
             Set of connections keys with more than one pair, so there is
-            the potential for conflicting times.
+            the potential ensure_conndict conflicting times.
```
Comment (bhazelton) [inline hera_mc/cm_health.py]: I'm a little confused by this docstring.

File: hera_mc/cm_health.py [hunk 3]
```diff
@@ -59,39 +67,37 @@ def ensure_conndict(self):
...
+            print('{} duplicate{} found.'.format(len(self.duplicates), s))
+            for dup in self.duplicates:
+                dc = self.conndict[dup[0]]
+                i0 = [dc[dup[1]].start_gpstime, dc[dup[1]].stop_gpstime]
+                i1 = [dc[dup[2]].start_gpstime, dc[dup[2]].stop_gpstime]
+                print('\t{}  --->  {} {}'.format(dc[0], i0, i1))
         else:
             print('No duplications found.')
-        print('Out of {} connections checked.'.format(self.num_connections))
+        print('{} connections checked.'.format(self.num_connections))
```
Comment (bhazelton) [inline hera_mc/cm_health.py]: Should it always print these messages, even if it's being called programmatically?

File: hera_mc/cm_health.py [hunk 4]
```diff
@@ -110,9 +116,11 @@ def check_for_existing_connection(self, connection, at_date='now'):
...
         if k not in self.conndict.keys():
             return False
-        for t in self.conndict[k]:
-            if cm_utils.is_active(at_date, t[0], t[1]):
-                print('Connection {} is already made for {}  ({} - {})'.format(k, at_date, t[0], t[1]))
+        for conn in self.conndict[k]:
+            t0 = conn.start_gpstime
+            t1 = conn.stop_gpstime
+            if cm_utils.is_active(at_date, t0, t1):
+                print('Connection {} is already made for {}  ({} - {})'.format(k, at_date, t0, t1))
```
Comment (bhazelton) [inline hera_mc/cm_health.py]: same comment -- do we always want these prints to happen?

**Comment by david-deboer** [comment]
Thanks - I addressed all of the issues.

**Comment by bhazelton** [comment]
Looks good, except that the coverage report says "* **6** unchanged lines in **3** files lost coverage." You can follow the link to the build, it shows which files & lines.

**Comment by david-deboer** [comment]
I didn't touch any of the files that are referenced in the "6 unchanged lines..." reference.

**Comment by bhazelton** [comment]
Ahh, it looks like all the changed lines are ones that are hit with only python 3, so coveralls must not have combined the different travis jobs properly.

**Pull request merged**